### ISSUE TITLE
.xml extension return 200 but should return 404

### ISSUE THREAD

**Issue by NicoHood** [getgrav/grav#3286]
As written in [this comment](https://github.com/getgrav/grav/issues/3130#issuecomment-803269679) page extensions with .xml still return 200

**Comment by mahagr** [comment]
Yeah, it's like that "by design". It is also something that needs to be fixed.

@rhukster Your thoughts?

https://github.com/getgrav/grav/blob/f78debe768758e98da3f641f41bab7005bb41807/system/src/Grav/Common/Twig/Twig.php#L451

**Comment by rhukster** [comment]
It is by design so that extra extensions can be used (such as `htm` or `xhtml`) and it will fall back to `.html` based Twig template .  I'm hesitant to change that behavior as it might have unintended consequences for existing sites which might be using that behavior in unexpected ways.

**Comment by NicoHood** [comment]
But that was recently fixed, that for example `.htm` or `.html` should redirect (301) instead of 200. Now `.xml` is a completely different type. It should not serve an html website.

**Comment by mahagr** [comment]
To be honest, I tend to agree with @NicoHood on this. The site should use a single extension (including no extension) and there is a configuration option for that. Everything else should either be 404 or 30x. It makes no sense to allow an infinite amount of file extensions just because someone may accidentally mistype the URL. That said, I cannot see anyone typing in URLs in the first place, which leaves only attackers who may want to try their luck.

**Comment by NicoHood** [comment]
Any updates on this?
